addons/road-generator/nodes/road_lane.gd

```diff
@@ -101,18 +101,19 @@ func register_vehicle(vehicle: Node) -> void:
 ## Optional but good cleanup of references.
 func unregister_vehicle(vehicle: Node) -> void:
 	if vehicle in _vehicles_in_lane:
+		print("Unregistering vehicle.")
 		_vehicles_in_lane.erase(vehicle)
 
 
 ## Return all vehicles registered to this lane, performing cleanup as needed.
 func get_vehicles() -> Array:
 	for vehicle in _vehicles_in_lane:
-		if not is_instance_valid(vehicle):
-			_vehicles_in_lane.erase(vehicle)
-			continue
-		if not vehicle or not vehicle._ai or vehicle._ai.follow_path != self:
+		if (not is_instance_valid(vehicle)) or vehicle.is_queued_for_deletion():
 			_vehicles_in_lane.erase(vehicle)
 			continue
+#		if not vehicle or not vehicle._ai or vehicle._ai.follow_path != self:
+#			_vehicles_in_lane.erase(vehicle)
+#			continue
```

Comment:
Suggesting to just remove the old code, seems this is done more logically in the brain lanes we use in Wheel Steal.

```suggestion
```